contracts/network/BancorNetwork.sol

```diff
@@ -878,6 +896,77 @@ contract BancorNetwork is IBancorNetwork, Upgradeable, ReentrancyGuardUpgradeabl
         _depositingEnabled = status;
     }
 
+    /**
+     * @inheritdoc IBancorNetwork
+     */
+    function feeExemptionWhitelist() external view returns (address[] memory) {
+        uint256 length = _feeExemptionWhitelist.length();
+        address[] memory list = new address[](length);
+        for (uint256 i = 0; i < length; i++) {
+            list[i] = _feeExemptionWhitelist.at(i);
+        }
+        return list;
+    }
+
+    /**
+     * @dev adds an address to the fee exemption whitelist
+     *
+     * requirements:
+     *
+     * - the caller must be the admin of the contract
+     */
+    function addToWhitelist(address addr) external onlyAdmin {
+        _addToWhitelist(addr);
+    }
+
+    /**
+     * @dev adds multiple addresses to the fee exemption whitelist
+     *
+     * requirements:
+     *
+     * - the caller must be the admin of the contract
+     */
+    function addAddressesToWhitelist(address[] calldata addrs) external onlyAdmin {
+        uint256 length = addrs.length;
+
+        for (uint256 i = 0; i < length; i++) {
+            _addToWhitelist(addrs[i]);
+        }
+    }
+
+    /**
+     * @dev adds an address to the fee exemption whitelist
+     */
+    function _addToWhitelist(address addr) private validExternalAddress(addr) {
```

Comment:
Solidity official style guide recommends putting all the internal and private functions after all the public and external functions

Reply:
fixed